crates/re_query/src/dataframe_util.rs

```diff
@@ -178,6 +230,46 @@ where
     }
 }
 
+impl<A: Archetype> ArchetypeView<A> {
+    pub fn as_df1<'a, C1: re_types::Component + Clone + Into<::std::borrow::Cow<'a, C1>> + 'a>(
+        &self,
+    ) -> crate::Result<DataFrame> {
+        let array0 =
+            re_types::components::InstanceKey::try_to_arrow(self.iter_instance_keys(), None)?;
+        let array1 = C1::try_to_arrow_opt(self.iter_optional_component::<C1>()?, None)?;
+
+        let series0 = Series::try_from((
+            re_types::components::InstanceKey::name().as_ref(),
+            array0.as_ref().clean_for_polars(),
+        ))?;
+        let series1 = Series::try_from((C1::name().as_ref(), array1.as_ref().clean_for_polars()))?;
+
+        Ok(DataFrame::new(vec![series0, series1])?)
+    }
+
+    pub fn as_df2<
+        'a,
+        C1: re_types::Component + Clone + Into<::std::borrow::Cow<'a, C1>> + 'a,
+        C2: re_types::Component + Clone + Into<::std::borrow::Cow<'a, C2>> + 'a,
+    >(
+        &self,
+    ) -> crate::Result<DataFrame> {
+        let array0 =
+            re_types::components::InstanceKey::try_to_arrow(self.iter_instance_keys(), None)?;
+        let array1 = C1::try_to_arrow_opt(self.iter_optional_component::<C1>()?, None)?;
+        let array2 = C2::try_to_arrow_opt(self.iter_optional_component::<C2>()?, None)?;
+
+        let series0 = Series::try_from((
+            re_types::components::InstanceKey::name().as_ref(),
+            array0.as_ref().clean_for_polars(),
+        ))?;
+        let series1 = Series::try_from((C1::name().as_ref(), array1.as_ref().clean_for_polars()))?;
+        let series2 = Series::try_from((C2::name().as_ref(), array2.as_ref().clean_for_polars()))?;
+
+        Ok(DataFrame::new(vec![series0, series1, series2])?)
+    }
```

Comment:
Those don't actually use the `df_builder_arch` things defined above :thinking: Anyway we can improve all of that later.